spec/event_store_client/client_spec.rb

```diff
@@ -47,6 +47,47 @@ module EventStoreClient
       end
     end
 
+    describe 'link_to' do
+      subject { -> { client.link_to(stream: stream_name, events: events) } }
+
+      let(:event_1) { Event.new(type: 'SomethingHappened', data: {}.to_json) }
+      let(:stream_name) { :stream_name }
+      let(:events) { [event_1] }
+
+      before do
+        allow_any_instance_of(Connection).to receive(:link_to).with(
+          stream_name,
+          events
+        ).and_return(events)
+      end
+
+      shared_examples 'argument error' do
+        it 'raises an Argument error' do
+          expect{ subject.call }.to raise_error(ArgumentError)
+        end
+      end
+
+      context 'when missing stream' do
+        let(:stream_name) { nil }
+        it_behaves_like 'argument error'
+      end
+
+      context 'when missing events' do
```

Comment:
Missing test for let(:events) { event } - it will crash with unexpected error (undefined_method :empty?)

Reply:
Added